### PR TITLE
Fix docs: Construct new TwoFactorAuth-class

### PR DESCRIPTION
**Pull request by modelrailroader**
Hey, 
your documentation to create a new object of the TwoFactorAuth class is wrong. 
A single string like "sha1" is not enough; expected is an argument of Algorithm.

### CODE REVIEW

**Review by willpower232 — CHANGES_REQUESTED** [review]
This is an excellent observation however technically one doesn't need to specify the other arguments now the minimum supported version of PHP is greater than 8 so this example could be `$tfa = new TwoFactorAuth(qrcodeprovider: $qrCodeProvider);`

That said, the documentation about algorithm at docs/optional-configuration.md is definitely out of date.

**Comment by modelrailroader** [comment]
Sorry, I really forget this. 
Do you think like this? I'm not so familiar with PHP 8 and all the changes yet...

**Comment by willpower232** [comment]
In this case it makes sense as I can imagine a significant number of users would just want to specify the name of the application and a QR code provider, unless one is developing for a specific hardware token or custom application then the default SHA1 should suffice which thanks to PHP 8 means you don't need to specify most of the constructor arguments.

Are you going to edit docs/optional-configuration.md as well?

**Comment by modelrailroader** [comment]
Done.

> In this case it makes sense as I can imagine a significant number of users would just want to specify the name of the application and a QR code provider

But wouldn't it then not making more sense to change the position of the CodeProvider, so that the time when a code gets unvalid and the number of digits don't have to specified when defining the class?

Like this:
`$tfa = new TwoFactorAuth(null, YourChosenProvider: $qrCodeProvider, 6, 30, Algorithm::Sha1);`

**Comment by modelrailroader** [comment]
I have done this now, like you proposed. 

Additionally, I fixed the hyperlinks for the documentation about the different online and offline providers. All hyperlinks had the file extension ".html", but it is ".md". ;-)

**Review by RobThree — APPROVED** [review]

File: docs/qr-codes.md [hunk 1]
```diff
@@ -49,14 +49,9 @@ If you do not want to use the default QR code provider, you can specify the one
 ```php
 use RobThree\Auth\TwoFactorAuth;
 
-$qrCodeProvider = new YourChosenProvider();
-
```
Comment (willpower232) [inline docs/qr-codes.md:53]: I think it is clearer if you put this back

**Review by willpower232 — CHANGES_REQUESTED** [review]

**Comment by modelrailroader** [comment]
You're right! Changed it.

**Review by willpower232 — APPROVED** [review]

**Comment by willpower232** [comment]
apologies for the delay! I wanted to do this at a time I could watch the build to make sure the docs website still works.

**Pull request merged**